runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6719,9 +6726,22 @@ TR_J9ByteCodeIlGenerator::genAconst_init(TR_OpaqueClassBlock *valueTypeClass)
                /// reference.
                if (fieldSignature[0] == 'Q')
                   {
+                  // In non-SVM AOT compilation, cpIndex is required for AOT relocation.
+                  // In this case, cpindex is unknown for the field.
+                  if (comp()->compileRelocatableCode() && !comp()->getOption(TR_UseSymbolValidationManager))
+                     {
+                     abortForUnresolvedValueTypeOp("aconst_init", "field");
+                     }
+
                   TR_OpaqueClassBlock *fieldClass = fej9()->getClassFromSignature(fieldSignature, (int32_t)strlen(fieldSignature),
                                                                                   comp()->getCurrentMethod());
-                  genAconst_init(fieldClass);
+                  if (comp()->getOption(TR_TraceILGen))
+                     {
+                     traceMsg(comp(), "fieldSignature %s fieldClass %p\n", fieldSignature, fieldClass);
+                     }
+
+                  // Set cpIndex as -1 since it's unknown for the field class
+                  genAconst_init(fieldClass, -1 /* cpIndex */);
```

Comment:
Because the parameter is a `uint16_t`, this -1 will be truncated to 65535. When the recursive call passes its `cpIndex` of 65535 to `findOrCreateClassSymbol()`, it will be zero-extended to `int32_t`, so the created symref will have a constant pool index of 65535 rather than the intended -1. The new `cpIndex` parameter should probably have type `int32_t` instead

Reply:
I realized I made the above mistake on this PR branch for the type of `cpIndex`. In my test branch, I had it changed to `int32_t`. Somehow I left it out on this PR branch. 

```
-   void         genAconst_init(uint16_t classCpIndex);
-   void         genAconst_init(TR_OpaqueClassBlock *valueTypeClass);
+   void         genAconst_init(int32_t classCpIndex);
+   void         genAconst_init(TR_OpaqueClassBlock *valueTypeClass, int32_t classCpIndex);
```

Reply:
Addressed in a59bddb

Reply:
The overload accepting only CP index should have been fine as it was, but it won't hurt for it to take `int32_t` either